tests/sles4sap/sap_deployment_automation_framework/cleanup.pm

```diff
@@ -13,26 +13,42 @@ use parent 'sles4sap::sap_deployment_automation_framework::basetest';
 use strict;
 use testapi;
 use warnings;
-use sles4sap::sap_deployment_automation_framework::deployment;
-use sles4sap::console_redirection;
+use sles4sap::sap_deployment_automation_framework::deployment
+  qw(serial_console_diag_banner
+  sdaf_cleanup
+  az_login
+  load_os_env_variables);
+use sles4sap::sap_deployment_automation_framework::deployment_connector qw(find_deployer_resources);
+use sles4sap::console_redirection qw(connect_target_to_serial disconnect_target_from_serial);
+use sles4sap::azure_cli qw(az_resource_delete);
 
 sub test_flags {
     return {fatal => 1};
 }
 
 sub run {
-    serial_console_diag_banner('end: sdaf_cleanup.pm');
+    serial_console_diag_banner('Start: sdaf_cleanup.pm');
     if (get_var('SDAF_RETAIN_DEPLOYMENT')) {
         record_info('Cleanup OFF', 'OpenQA variable "SDAF_RETAIN_DEPLOYMENT" is active, skipping cleanup.');
         return;
     }
 
-    # Cleanup SDAF files form Deployer VM
+    # Trigger SDAF remover script and clean up files from deployer VM
     connect_target_to_serial();
     load_os_env_variables();
     az_login();
     sdaf_cleanup();
     disconnect_target_from_serial();
+
+    # Cleanup deployer VM resources
```

Comment:
This code is same as the post_fail_hook: maybe move it in a function.

Reply:
Yeah. I will deal with that in the next ticket TEAM-9459 which is specifically to do the cleanup part. 
Cleanup I included here is only a simple one to have at least something. 

I'll probably convert this to a function but I am kinda torn in there since it makes test modules lack transparency especially to the reviewer. You just see some functions being called and then you have to dig through github dirs. But well... it is what it is.